File: src/main/java/io/github/colintimbarndt/chat_emotes/data/unicode/joiner/IdentityTransformer.java
package io.github.colintimbarndt.chat_emotes.data.unicode.joiner;

import io.github.colintimbarndt.chat_emotes.data.unicode.pattern.MatchResult;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public final class IdentityTransformer implements MatchTransformer {
    public static final IdentityTransformer INSTANCE = new IdentityTransformer();
    private IdentityTransformer() {}
    @Override
    public void transform(@NotNull MatchResult r, @NotNull StringBuilder result) {
        result.append(Objects.requireNonNull(r.name()));
    }

    @Override
    public boolean acceptsWidth(int w) {
        return true;
    }

    @Override
    public int acceptedWidth() {
        return 0;
    }

    @Override
    public boolean requiresName() {
        return true;
    }
}

File: common/src/main/kotlin/io/github/colintimbarndt/chat_emotes/common/data/unicode/joiner/IdentityTransformer.kt
package io.github.colintimbarndt.chat_emotes.common.data.unicode.joiner

import io.github.colintimbarndt.chat_emotes.common.data.unicode.pattern.MatchResult
import java.util.*

object IdentityTransformer : MatchTransformer {

    override fun transform(r: MatchResult, result: StringBuilder) {
        result.append(Objects.requireNonNull(r.name))
    }

    override fun acceptsWidth(w: Int) = true

    override val acceptedWidth = 0

    override val requiresName = true

}